src/Core/Services/UsernameGenerationService.cs

```diff
@@ -0,0 +1,250 @@
+                }
+                else
+                {
+                    options.Merge(_defaultOptions);
+                    _optionsCache = options;
+                }
+            }
+
+            return _optionsCache;
+        }
+        public async Task SaveOptionsAsync(UsernameGenerationOptions options)
+        {
+            await _stateService.SetUsernameGenerationOptionsAsync(options);
+            _optionsCache = options;
+        }
+
+        public void ClearCache()
+        {
+            _optionsCache = null;
+        }
+
+        private async Task<string> GenerateRandomWordAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string randomWord = null;
+            if (options.RandomWordUsernameCapitalize == null)
+            {
+                options.RandomWordUsernameCapitalize = false;
+            }
+            if (options.RandomWordUsernameIncludeNumber == null)
+            {
+                options.RandomWordUsernameIncludeNumber = false;
+            }
+            var listLength = EEFLongWordList.Instance.List.Count - 1;
+            var wordIndex = await _cryptoService.RandomNumberAsync(0, listLength);
+            if (options.RandomWordUsernameCapitalize.GetValueOrDefault())
+            {
+                randomWord = Capitalize(EEFLongWordList.Instance.List[wordIndex]);
+            }
+            else
+            {
+                randomWord = EEFLongWordList.Instance.List[wordIndex];
+            }
+
+            if (options.RandomWordUsernameIncludeNumber.GetValueOrDefault())
+            {
+                randomWord = await AppendRandomNumberToRandomWordAsync(randomWord);
+            }
+
+            return randomWord;
+        }
+
+        private async Task<string> GeneratePlusAddressedEmailAsync(UsernameGenerationOptions options)
+        {
+            options.Merge(_defaultOptions);
+            string generatedString = string.Empty;
+            var adressedEmail = options.PlusAddressedEmail;
+
+            if (string.IsNullOrWhiteSpace(adressedEmail) || adressedEmail.Length < 3)
+            {
+                return DefaultGenerated;
+            }
+
+            var atIndex = adressedEmail.IndexOf("@");
+            if (atIndex < 1 || atIndex >= adressedEmail.Length - 1)
+            {
+                return adressedEmail;
+            }
+
+            var emailBeginning = adressedEmail.Substring(0, atIndex);
+            var emailEnding = string.Empty;
+
+            emailEnding = adressedEmail.Substring(atIndex + 1, adressedEmail.Length - (atIndex + 1));
+
+            if (options.PlusAddressedEmailType == UsernameEmailType.Random)
+            {
+                generatedString = await RandomStringAsync(8);
+            }
+            else
+            {
+                generatedString = options.EmailWebsite;
+            }
+
+            return emailBeginning + "+" + generatedString + "@" + emailEnding;
```

Comment:
This can be easier with `Insert`, so that you don't need the beginning nor the ending nor the generatedString:
```c#
if (options.PlusAddressedEmailType == UsernameEmailType.Random)
{
    var randomString = await RandomStringAsync(8);
    return options.PlusAddressedEmail.Insert(atIndex, $"+{randomString}");
}
else
{
    return options.PlusAddressedEmail.Insert(atIndex, $"+{options.EmailWebsite}");
}
```